### PR TITLE
SunOS additions and fixes

### PR DESCRIPTION
**Pull request by pieterlexis**
Also, added service definition files for the auth and recursor

### CODE REVIEW

**Comment by rubenk** [comment]
LGTM!

File: m4/pdns_enable_reproducible.m4 [hunk 1]
```diff
@@ -12,7 +12,9 @@ AC_DEFUN([PDNS_ENABLE_REPRODUCIBLE], [
     AC_DEFINE([REPRODUCIBLE], [1], [Define to 1 for reproducible builds])
   ],[
     build_user=m4_esyscmd_s(id -u -n)
-    build_host=m4_esyscmd_s(hostname -f || hostname)
+    AS_IF([test x"$host_os" = "xSunOS"],[
+      build_host=m4_esyscmd_s(uname -srvp)],[
```
Comment (Habbie) [inline m4/pdns_enable_reproducible.m4]: this returns `SunOS 5.9 Generic_118558-39 sparc` which does not look like a hostname

File: pdns/configure-recursor [hunk 2]
```diff
@@ -26,8 +26,17 @@ CF_PIE=""
 LD_PIE=""
 CF_FORTIFY=""
 CF_STACK=""
-BUILD_HOST="$(id -u -n)@$(hostname -f 2>/dev/null || hostname 2>/dev/null || echo localhost)"
 
+BUILD_USER="$(id -u -n)"
+case $(uname) in
+  SunOS)
+    BUILD_HOST="${BUILD_USER}@$(uname -srvp)" ;;
```
Comment (Habbie) [inline pdns/configure-recursor]: this returns `SunOS 5.9 Generic_118558-39 sparc` which does not look like a hostname

**Comment by Habbie** [comment]
No contest, feel free to pretend LGTM :)

**Pull request merged**